src/struct_info.rs

```diff
@@ -62,14 +62,16 @@ impl<'a> StructInfo<'a> {
             ..
         } = *self;
         let (impl_generics, ty_generics, where_clause) = self.generics.split_for_impl();
-        let all_fields_param =
-            syn::GenericParam::Type(syn::Ident::new("TypedBuilderFields", proc_macro2::Span::call_site()).into());
+        let empties_tuple = type_tuple(self.included_fields().map(|_| empty_type()));
+        let mut all_fields_param_type: syn::TypeParam =
+            syn::Ident::new("TypedBuilderFields", proc_macro2::Span::call_site()).into();
+        let all_fields_param = syn::GenericParam::Type(all_fields_param_type.clone());
         let b_generics = self.modify_generics(|g| {
-            g.params.insert(0, all_fields_param.clone());
+            all_fields_param_type.default = Some(syn::Type::Tuple(empties_tuple.clone()));
```

Comment:
Why does this need to be inside `modify_generics`?

Reply:
I just need an answer for that before I merge. I can move it outside after the merge, but I really don't want to leave it like that and forget about it...

Reply:
I was not sure what it might change if I do it outside, as it is reused in other places. Let me see if I can improve on this and see if it actually hurts. This is the part I meant when talking about it not being clean yet.

Reply:
Ah I see what you mean now. It does not need to be inside the closure, but it definitely needs to be after it was used to construct the `GenericParam::Type`. I am still not really happy with it, but outside is better.

Reply:
After I merge this PR, I think I'll just get rid of `modify_generics`. It was added by a very incompetent programmer (2018's me) and is not helpful enough to justify the confusion it sows.